Terminal.Gui/View/ViewScrollBar.cs

```diff
@@ -0,0 +1,635 @@
+﻿namespace Terminal.Gui;
+
+/// <summary>
+///     The scroll bar types used by this <see cref="ScrollBarView"/>.
+/// </summary>
+public enum ScrollBarType
+{
+    /// <summary>
+    ///     None scroll bar will be used and to avoid throwing an exception never use it in a constructor.
+    /// </summary>
+    None,
+
+    /// <summary>
+    ///     Only the vertical scroll bar will be shown.
+    /// </summary>
+    Vertical,
+
+    /// <summary>
+    ///     Only the horizontal scroll bar will be shown.
+    /// </summary>
+    Horizontal,
+
+    /// <summary>
+    ///     Both vertical and horizontal scroll bars will be shown.
+    /// </summary>
+    Both
+}
+
+public partial class View
+{
+    private ScrollBarView _scrollBar;
+    private ScrollBarType _scrollBarType;
+    private int _scrollColsSize;
+    private int _scrollLeftOffset;
+    private int _scrollRowsSize;
+    private int _scrollTopOffset;
+
+    /// <summary>If true the vertical/horizontal scroll bars won't be showed if it's not needed.</summary>
+    public bool ScrollAutoHideScrollBars
+    {
+        get => _scrollBar.AutoHideScrollBars;
+        set => _scrollBar.AutoHideScrollBars = value;
+    }
+
+    /// <summary>
+    ///     Gets or sets the <see cref="ScrollBarType"/> used by this view.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException">The value is out of range.</exception>
+    public virtual ScrollBarType ScrollBarType
+    {
+        get => _scrollBarType;
+        set
+        {
+            if (_scrollBar is { } && _scrollBarType == value)
+            {
+                return;
+            }
+
+            _scrollBarType = value;
+            DisposeScrollBar ();
+
+            switch (_scrollBarType)
+            {
+                case ScrollBarType.Vertical:
+                    _scrollBar = new ScrollBarView { IsVertical = true };
+
+                    break;
+                case ScrollBarType.Horizontal:
+                    _scrollBar = new ScrollBarView { IsVertical = false };
+
+                    break;
+                case ScrollBarType.Both:
+                    _scrollBar = new ScrollBarView { IsVertical = true };
+                    _scrollBar.OtherScrollBarView = new ScrollBarView { IsVertical = false, OtherScrollBarView = _scrollBar };
+
+                    break;
+                case ScrollBarType.None:
+                    return;
+                default:
+                    throw new ArgumentOutOfRangeException ();
+            }
+
+            Padding.Add (_scrollBar);
+            AddHandlers ();
+            AddScrollKeyBindings (_scrollBar);
+
+            if (_scrollBar.OtherScrollBarView != null)
+            {
+                AddScrollKeyBindings (_scrollBar.OtherScrollBarView);
+            }
+
+            SetNeedsDisplay ();
+        }
+    }
+
+    /// <summary>
+    ///     Determines the number of columns to scrolling.
+    /// </summary>
+    public int ScrollColsSize
```

Comment:
Neither of them. It is the number of columns that a view handles (it could be the length of a string or the number of columns in a table, like the `TableView` is using). If this length plus the left offset exceeds the dimensions (`Bounds.Width`), then the scroll bar will adjust it accordingly.
However, the description really isn't correct, so I ask you to let me know what description you want me to give. Thanks.